### PR TITLE
fix(scheduler): fix pod to claim connection calculation

### PR DESCRIPTION
**Pull request by davidLif**
## Description

Bug fix - in the GetDraPodClaims, connect the claim using ResourceClaimName, and not the name field in PodResourceClaim.

Validate fix unitest - "should skip claims with nil ResourceClaimName - even if a template name is provided"

## Related Issues

Fixes #

## Checklist

> **Note:** Ensure your PR title follows the [Conventional Commits format](https://github.com/NVIDIA/KAI-Scheduler/blob/main/CONTRIBUTING.md#pr-title-guidelines) (e.g., `feat(scheduler): add new feature`)

- [x] Self-reviewed
- [x] Added/updated tests (if needed)
- [ ] Updated documentation (if needed)

## Summary by CodeRabbit

* **New Features**
  * Enhanced Dynamic Resource Allocation support in Kubernetes pod scheduling through improved resource claim handling and pod-to-claim associations.

* **Tests**
  * Added comprehensive test coverage for Dynamic Resource Allocation resource utilities, including edge cases and multiple allocation scenarios.

### CODE REVIEW

**Comment by davidLif** [comment]
@coderabbitai review

**Review by enoodle — DISMISSED** [review]

File: pkg/scheduler/api/resource_info/dra_resource_utils.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	PodOwnerKind         string = "Pod"
+	ReservedForPodPlural string = "pods"
+)
+
+func ResourceClaimSliceToMap(draResourceClaims []*resourceapi.ResourceClaim) map[string]*resourceapi.ResourceClaim {
+	draClaimMap := map[string]*resourceapi.ResourceClaim{}
+	for _, draClaim := range draResourceClaims {
+		draClaimMap[draClaim.Name] = draClaim
+	}
+	return draClaimMap
```
[the remark this answers is not in the document]
Comment (davidLif) [reply pkg/scheduler/api/resource_info/dra_resource_utils.go:22]: fixed

**Review by enoodle — APPROVED** [review]
